gsy-orderbook-service/src/event_listener.rs

```diff
@@ -1,73 +1,79 @@
 use crate::db::DbRef;
 use crate::schema::OrderStatus;
 use anyhow::{Error, Result};
-use futures::StreamExt;
 use mongodb::bson;
-use subxt::{ClientBuilder, DefaultConfig, SubstrateExtrinsicParams};
+use subxt::{
+    SubstrateConfig,
+    OnlineClient,
+};
 use tracing::info;
 
 #[subxt::subxt(runtime_metadata_path = "metadata.scale")]
 pub mod gsy_node {}
 
 pub async fn init_event_listener(db: DbRef, node_url: String) -> Result<(), Error> {
-    let api = ClientBuilder::new()
-        .set_url(format!("ws://{}", node_url))
-        .build()
-        .await?
-        .to_runtime_api::<gsy_node::RuntimeApi<DefaultConfig, SubstrateExtrinsicParams<DefaultConfig>>>();
 
-    let mut gsy_node_events = api.events().subscribe().await?.filter_events::<(
-        gsy_node::orderbook_registry::events::OrderExecuted,
-        gsy_node::orderbook_registry::events::OrderDeleted,
-    )>();
+    let api = OnlineClient::<SubstrateConfig>::from_url(format!("ws://{}", node_url)).await?;
 
-    while let Some(events) = gsy_node_events.next().await {
-        let events = events?;
-        let block_hash = events.block_hash;
-        let event = events.event;
-        info!("Events at block {:?}:", block_hash);
+    let mut gsy_node_events = api.blocks().subscribe_all().await?;
+
+    while let Some(block) = gsy_node_events.next().await {
+        let block = block?;
+
+        // Ask for the events for this block.
+        let events = block.events().await?;
 
-        if let (Some(order_executed), _) = &event {
-            info!("Order Executed: {:?}", order_executed);
+        let block_hash = block.hash();
+        // let events = events?;
+        // let block_hash = events.block_hash;
+        // let event = events.event;
+        info!("Events at block {:?}:", block_hash);
+        for event in events.find::<gsy_node::orderbook_registry::events::OrderExecuted>() {
+            if let Ok(order_executed) = &event {
+                info!("Order Executed: {:?}", order_executed);
 
-            let id = &bson::to_bson(&order_executed.0.offer_hash).unwrap();
-            match db
-                .get_ref()
-                .orders()
-                .update_order_status_by_id(id, OrderStatus::Executed)
-                .await
-            {
-                Ok(result) => info!("Update result: {:?}", result),
-                Err(e) => {
-                    tracing::error!("Failed to execute update: {:?}", e);
+                let id = &bson::to_bson(&order_executed.0.offer_hash).unwrap();
```

Comment:
Do I get it right, that this is the place where the trade is written to mongo?

Reply:
Exactly, this changes the status of the order from Open to Executed.